### PR TITLE
#167484599 Users should be able to receive notifications

### PR DESCRIPTION
**Pull request by harerakalex**
### What does this PR do?
- Users should be able to receive notifications
- Users should be able to receive email notifications
- Users should be able to receive realtime in app notification  
### Description of Task to be completed?
- Create new controller for notifications
- Create helper for article notification
- Create socket file to provide in app real time notification
- Create test file to test this feature
- Document feature using swagger
### How should this be manually tested?
- clone this branch
- npx sequelize-cli db:migrate
- Run npm start
- Notification is sent after creating new article 
- Use this route `http://localhost:3000/api/v1/profiles/:username/notification/unsubscribe` to unsubscribe to the email notification.
- Use this route `http://localhost:3000/api/v1/profiles/notification/all` to get all notifications of a specific users.
- Use this route `http://localhost:3000/api/v1/profiles/notification/:id/read` to mark a specific notification as read
- Use this route `http://localhost:3000/api/v1/profiles/notification/read/all` to mark all notifications as read.
- N.B: - All those routes are protected so use header `Authorization`
           - Notification is received by user who follows a publisher of an article. 
### New introduced environment variables
- SOCKET_PORT = 5000 
- APP_URL_FRONTEND = "http://localhost:3000/api/v1"
### Any background context you want to provide?
- N/A
### What are the relevant pivotal tracker stories?
[Users should be able to receive notifications](https://www.pivotaltracker.com/story/show/167484599)
### Questions
- N/A
### Screenshot
- email notification
Screen Shot 2019-08-21 at 21 52 09

### CODE REVIEW

**Comment by harerakalex** [comment]
New introduced environment variables for this feature are:
SOCKET_PORT = 5000 
APP_URL_FRONTEND = "http://localhost:3000/api/v1"

**Review by malaba6 — CHANGES_REQUESTED** [review]
You forgot to add Herman @mkiterian as a reviewer

**Comment by harerakalex** [comment]
> You forgot to add Herman @mkiterian as a reviewer

Thank you @Malaba6

**Review by placideirandora — APPROVED** [review]

File: src/routes/routes.js [hunk 1]
```diff
@@ -45,4 +46,9 @@ router.delete('/api/v1/profiles/:userName/unfollow', verifyToken, UserFollow.unF
 router.get('/api/v1/profiles/:userName/following', verifyToken, UserFollow.getFollowingList);
 router.get('/api/v1/profiles/:userName/followers', verifyToken, UserFollow.getFollowersList);
 
+router.patch('/api/v1/profiles/:username/notification/:subscribe', [verifyToken, canEditProfile], Notification.optInOutNotificatation);
```
Comment (mkiterian) [inline src/routes/routes.js]: by convention, the resource part has to be in plural i.e. `notifications` not `notification`. Otherwise looks good

**Review by mkiterian — CHANGES_REQUESTED** [review]
Looks good, works as expected. One minor thing to change

File: src/routes/routes.js (hunk 1 above)
Comment (harerakalex) [reply src/routes/routes.js]: Thank you, that's right.

Comment (harerakalex) [reply src/routes/routes.js]: Done

**Pull request merged**